src/Domain/Creators/Shell.cpp

```diff
@@ -92,15 +102,17 @@ Domain<3> Shell::create_domain() const noexcept {
   std::vector<DirectionMap<
       3, std::unique_ptr<domain::BoundaryConditions::BoundaryCondition>>>
       boundary_conditions_all_blocks{};
+
+  const size_t blocks_per_layer =
```

Comment:
why is this moved outside the loop? isn't the number of blocks equal to the size of coord_maps?